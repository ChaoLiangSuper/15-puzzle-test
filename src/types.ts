export interface IAppState {
  tiles: Number[];
  isSolved: boolean;
}

export interface ICellProp {
  displayValue: number;
  handleCellclick: () => void;
}

export interface IGridProp {
  tiles: number[];
  edgeLength: number;
  handleCellclick: (p: number) => void;
  isSolved: boolean;
}

export interface IOverlayProps {
  displayText: string;
}
