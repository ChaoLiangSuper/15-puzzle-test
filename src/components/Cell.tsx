import React from 'react';
import { ICellProp } from '../types';

const Cell: React.FC<ICellProp> = ({ displayValue, handleCellclick }) => {
  const className = displayValue === 0 ? 'cell cell-empty' : 'cell';
  return (
    <div className={className} onClick={handleCellclick}>
      {displayValue === 0 ? null : displayValue}
    </div>
  );
};

export default Cell;
