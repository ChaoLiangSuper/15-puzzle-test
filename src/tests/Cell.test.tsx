import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Cell from '../components/Cell';
import { ICellProp } from '../types';

describe('<Cell />', () => {
  let wrapper: ShallowWrapper<ICellProp>;
  let props: ICellProp;

  beforeEach(() => {
    props = {
      displayValue: 5,
      handleCellclick: jest.fn(),
    };
    wrapper = shallow(<Cell {...props} />);
  });

  it('should initialize without crash and match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to get click event', () => {
    expect(props.handleCellclick).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(props.handleCellclick).toHaveBeenCalledTimes(1);
  });

  it('should render empty cell for empty tile', () => {
    wrapper.setProps({
      ...props,
      displayValue: 0,
    });

    expect(wrapper.find('div').props().className).toBe('cell cell-empty');
    expect(wrapper.text()).toBe('');
  });
});
