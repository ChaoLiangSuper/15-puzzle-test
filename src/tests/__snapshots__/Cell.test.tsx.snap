// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> should initialize without crash and match snapshot 1`] = `
<div
  className="cell"
  onClick={[MockFunction]}
>
  5
</div>
`;
